src/components/MainComponent/Description/Description.js

```diff
@@ -1,58 +1,107 @@
-import "./Description.scss"
+import React, { Component } from 'react';
+import "./Description.scss";
+import "../../../api/api"
+import { buttonsName, foodData } from "../../../constants/constants";
+import Nutrition from "../Nutrition/Nutrition";
+import FoodCard from "../FoodCards/FoodCard";
+import {getData} from "../../../api/api";
 
-import {buttonsName, foodData} from "../../../constants/constants"
+class Description extends Component {
 
-import Nutrition from "../Nutrition/Nutrition";
-import FoodCards from "../FoodCards/FoodCards";
+    constructor(props) {
+        super(props);
+        this.state = {
+            meals: [],
+            mealsForShow: [],
+            isShow: true
+        };
+    }
 
-const Description = () => {
-    const mouseHover = () => {
-        document.querySelector('.hint').classList.remove('hidden')
+    componentDidMount() {
+        getData().then(data => {
+             this.setState({ meals: data });
+             this.setState({mealsForShow: data.slice(0,6)})
+        })
     }
-    const mouseLeave = () => {
-        document.querySelector('.hint').classList.add('hidden')
+
+    mouseHover = () => {
+        document.querySelector('.hint').classList.remove('hidden');
+    };
+
+    mouseLeave = () => {
+        document.querySelector('.hint').classList.add('hidden');
+    };
+
+    handlePagination = () => {
+        const isMoreThanSix =  this.state.meals.length - this.state.mealsForShow.length >= 6;
+        const offset = this.state.meals.length - this.state.mealsForShow.length;
+        if (offset > 1) {
+            const mealsForShowLength = this.state.mealsForShow.length;
+            let newMealsForShow;
+            if (isMoreThanSix) {
+                newMealsForShow = this.state.mealsForShow.concat(this.state.meals.slice(mealsForShowLength, mealsForShowLength + 6));
```

Comment:
We can't modify the state anyhow except the setState method. You need to remember that the setState method accepts a callback witch takes the prevState and returns the new one. All current logic should be placed inside this callback.